packages/react-native-editor/src/setup-locale.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * WordPress dependencies
+ */
+import { setLocaleData } from '@wordpress/i18n';
+
+/**
+ * Setup locale data for default domain and plugins.
+ *
+ * @typedef {Object} PluginTranslation
+ * @property {string}              domain                Domain of the plugin.
+ * @property {Function}            getTranslation        Function for retrieving translations for a locale.
+ *
+ * @param    {string}              locale                Locale value.
+ * @param    {Object}              extraTranslations     Extra translations to be included.
+ * @param    {Function}            getDefaultTranslation Default domain's function for retrieving translations for a locale.
+ * @param    {PluginTranslation[]} pluginTranslations    Array with plugin translations.
+ */
+export default (
+	locale,
+	extraTranslations,
+	getDefaultTranslation,
+	pluginTranslations
+) => {
+	const setDomainLocaleData = ( getTranslation, domain ) => {
+		let translations = getTranslation( locale );
+		if ( locale && ! translations ) {
+			// Try stripping out the regional
+			locale = locale.replace( /[-_][A-Za-z]+$/, '' );
+			translations = getTranslation( locale );
+		}
+		const allTranslations = Object.assign(
+			{},
+			translations,
+			extraTranslations
+		);
+		// eslint-disable-next-line no-console
+		console.log(
+			domain ? `${ domain } - locale` : 'locale',
+			locale,
+			allTranslations
+		);
+		// Only change the locale if it's supported by gutenberg
+		if ( translations || extraTranslations ) {
+			setLocaleData( allTranslations, domain );
+		}
+	};
+
+	// Set up default locale data (Gutenberg)
+	setDomainLocaleData( getDefaultTranslation );
+
+	// Set up plugin translations
+	pluginTranslations.forEach( ( { domain, getTranslation } ) =>
+		setDomainLocaleData( getTranslation, domain )
+	);
```

Comment:
Presenting this as smaller alternative, but feel free to disregard if it feels too clever. 

<details><summary>Alternative Code</summary>

```js
const setDomainLocaleData = ( { getTranslation, domain = 'default' } ) => {
  let translations = getTranslation( locale );
  if ( locale && ! translations ) {
    // Try stripping out the regional
    locale = locale.replace( /[-_][A-Za-z]+$/, '' );
    translations = getTranslation( locale );
  }
  const allTranslations = {
    ...translations,
    ...extraTranslations,
  };
  // eslint-disable-next-line no-console
  console.log( `>>> ${ domain } - locale`, locale, allTranslations );
  // Only change the locale if it's supported by gutenberg
  if ( translations || extraTranslations ) {
    setLocaleData( allTranslations, domain );
  }
};

// Set up default local and plugin translations
[
  { getTranslation: getDefaultTranslation },
  ...pluginTranslations,
].forEach( setDomainLocaleData );
```

</details>

Reply:
Thanks for sharing this alternative 🙇 ! I like the idea of unifying the call with `forEach`.

Reply:
I applied the suggestion in [this commit](https://github.com/WordPress/gutenberg/pull/37073/commits/f939c13e7e0eb9549732503c9e89dff58fc2f4e5).